rclpy/rclpy/context.py

```diff
@@ -46,7 +47,11 @@ def __init__(self):
     def handle(self):
         return self._handle
 
-    def init(self, args: Optional[List[str]] = None, *, initialize_logging: bool = True):
+    def init(self,
+             args: Optional[List[str]] = None,
+             *,
+             initialize_logging: bool = True,
+             domain_id: int = rclpy_implementation.rclpy_get_default_domain_id()):
```

Comment:
why call implementation instead of calling utilities.get_default_domain_id?

Reply:
It is because utilities.py has imported context.py.
https://github.com/Barry-Xu-2018/rclpy/blob/02d0bb6aa33b8e44a97ecac6cae4bbb8fce20f34/rclpy/rclpy/utilities.py#L22  

In order to avoid interdependence, have to call rclpy_implementation.